src/web/Image.tsx

```diff
@@ -154,6 +154,30 @@ export class Image extends React.Component<Types.ImageProps, ImageState> {
         return defer.promise();
     }
 
+    static getMetadata(url: string): SyncTasks.Promise<Types.ImageMetadata> {
+        const defer = SyncTasks.Defer<Types.ImageMetadata>();
+
+        const img = new (window as any).Image();
+        img.src = url;
+
+        img.onload = ((event: Event) => {
```

Comment:
Are onload, onerror and onabort guaranteed to be called even if the operation completes before the handlers are specified? If not, we should move "img.src = url" further down so the operation isn't started until after the handlers are installed.